### PR TITLE
Feature-3533 Adding new Istio support

### PR DESCRIPTION
**Pull request by mishah334**
**Updates**

- Adding Istio firewall rule.
- Adding changes to support the new Terraform version and providers.
- Updating `pre-commit` hooks.
- Updating CI to use `pre-commit`.

Issue: https://github.com/astronomer/issues/issues/3533

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,27 +1,71 @@
 ---
 version: 2.1
+
 orbs:
   slack: circleci/slack@4.1.3
+
+executors:
+  pre-commit-executor:
+    docker:
+      - image: quay.io/astronomer/ci-pre-commit:2022-06
```
Comment (danielhoherd) [inline .circleci/config.yml:10]: Might as well keep this fresh.

```suggestion
      - image: quay.io/astronomer/ci-pre-commit:2022-08
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,27 +1,71 @@
...
+        environment:
+          TFENV_AUTO_INSTALL: true
+  terraform-executor:
+    docker:
+      - image: quay.io/astronomer/ci-terraform:2022-06
```
Comment (danielhoherd) [inline .circleci/config.yml:15]: Might as well keep this fresh 2

```suggestion
      - image: quay.io/astronomer/ci-pre-commit:2022-08
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,27 +1,71 @@
...
+      #- git_tag:
+      #    context:
+      #      - github-repo
+      #      - slack
+      #    requires:
+      #      - terraform_destroy
+      #    filters:
+      #      branches:
+      #        only:
+      #          - master
```
Comment (danielhoherd) [inline .circleci/config.yml:49]: What's the story here? Are we going to uncomment this? If not, let's delete it.

File: .circleci/config.yml [hunk 4]
```diff
@@ -102,34 +144,38 @@ jobs:
...
-      #    context:
-      #      - github-repo
-      #      - slack
-      #    requires:
-      #      - terraform_destroy
-      #    filters:
-      #      branches:
-      #        only:
-      #          - master
+commands:
```
Comment (danielhoherd) [inline .circleci/config.yml:147]: I've been kinda anti-circleci-commands, but actually i think this use of them makes things cleaner. Very cool.

File: locals.tf [hunk 5]
```diff
@@ -22,28 +22,28 @@ locals {
...
+  #current-context: "context"
+  #kind: Config
+  #preferences: {}
+  #users:
+  #- name: "${google_container_cluster.primary.master_auth[0].username}"
+  #  user:
+  #    password: "${google_container_cluster.primary.master_auth[0].password}"
+  #    username: "${google_container_cluster.primary.master_auth[0].username}"
+  #
+  #  EOF
```
Comment (danielhoherd) [inline locals.tf:46]: Can we just delete this? If we need it back we can grab it from git history.

File: main.tf [hunk 6]
```diff
@@ -143,10 +141,10 @@ resource "random_id" "kubeconfig_suffix" {
...
 }
 
-resource "local_file" "kubeconfig" {
-  sensitive_content = local.kubeconfig
-  filename          = "./kubeconfig-${random_id.kubeconfig_suffix.hex}"
-}
+#resource "local_file" "kubeconfig" {
+#  sensitive_content = local.kubeconfig
+#  filename          = "./kubeconfig-${random_id.kubeconfig_suffix.hex}"
+#}
```
Comment (danielhoherd) [inline main.tf:147]: This can also be recovered from git history so it can be deleted.

File: outputs.tf [hunk 7]
```diff
@@ -57,14 +57,14 @@ EOF
...
 }
 
-output "kubeconfig" {
-  value     = local.kubeconfig
-  sensitive = true
-}
+#output "kubeconfig" {
+#  value     = local.kubeconfig
+#  sensitive = true
+#}
```
Comment (danielhoherd) [inline outputs.tf:63]: Delete, can be recovered from git history.

File: outputs.tf [hunk 8]
```diff
@@ -57,14 +57,14 @@ EOF
...
 
-output "kubeconfig_filename" {
-  value = local_file.kubeconfig.filename
-}
+#output "kubeconfig_filename" {
+#  value = local_file.kubeconfig.filename
+#}
```
Comment (danielhoherd) [inline outputs.tf:67]: Delete, can be recovered from git history.

**Review by danielhoherd — COMMENTED** [review]
Looks good! Got a few cleanup suggestions but no show stoppers.

**Review by danielhoherd — DISMISSED** [review]

**Review by santoshr1016 — APPROVED** [review]
GTG

**Pull request merged**